pilot/docker/Dockerfile.pilot

```diff
@@ -10,7 +10,7 @@ COPY cacert.pem /cacert.pem
 # 1001 is chosen as the UID to be added to the istio group as this is the UID that pilot-discovery in the docker container runs as.
 RUN addgroup --system --gid 1001 istio && \
```

Comment:
The reasons for not calling addgroup/adduser in this docker image:
- inefficient - just add it to the base image, no need to run it again and again
- we may want to switch to better non-priv-required builder for docker images. They work great for adding files, but are trickier for RUN. Most of our dockers don't really need RUN.

Reply:
@costinm this adduser stuff is orthogonal to the problem this PR addresses.  The actual problem is that binaries have the wrong permissions afaik from the PR contents.

Can you elaborate on the non-priv-required builder?  Not familiar with what you are referencing?

Reply:
@costinm some images are built from scratch instead of `base_debug`. That is why I adduser here. If do it in base image, we need use `base_debug` for all images.

Reply:
@costinm @sdake  I just changed the base image to add `istio-proxy` user.  it seems that it needs to have base image pushed firstly, then the test can pass. but I cannot push it without testing. so it is a little contradiction. Is my understanding wrong?